ui/src/wallet.ts

```diff
@@ -0,0 +1,259 @@
+import WalletConnect from "@walletconnect/client";
+import * as svelteStore from "svelte/store";
+import * as ethers from "ethers";
+import * as ethersBytes from "@ethersproject/bytes";
+import {
+  Deferrable,
+  defineReadOnly,
+  resolveProperties,
+} from "@ethersproject/properties";
+import type {
+  Provider,
+  TransactionRequest,
+} from "@ethersproject/abstract-provider";
+
+import * as contract from "../src/funding/contract";
+import * as modal from "../src/modal";
+import * as path from "../src/path";
+import * as notification from "../src/notification";
+
+export enum Status {
+  Connected = "CONNECTED",
+  Connecting = "CONNECTING",
+  NotConnected = "NOT_CONNECTED",
+}
+
+export type State =
+  | { status: Status.NotConnected; error?: Error }
+  | { status: Status.Connecting }
+  | { status: Status.Connected; connected: Connected };
+
+export interface Connected {
+  account: Account;
+}
+
+export interface Account {
+  address: string;
+  balance: string;
+}
+
+export interface Wallet extends svelteStore.Readable<State> {
+  connect(): Promise<void>;
+  disconnect(): Promise<void>;
+  signer: ethers.Signer;
+}
+
+export const provider = new ethers.providers.JsonRpcProvider(
+  "http://localhost:8545"
+);
+
+export function build(): Wallet {
```

Comment:
I think I have tried that before. If my memory serves, that didn't quite work because `class` cannot be serialized and stored in the svelte store. Will have a look cause indeed this is huge and super hard to follow. :+1:

Reply:
Pretty much the same answer here as https://github.com/radicle-dev/radicle-upstream/pull/975#discussion_r555060510.

Reply:
I’m fine with it